xcm/src/v1/multiasset.rs

```diff
@@ -0,0 +1,484 @@
+// Copyright 2020 Parity Technologies (UK) Ltd.
+// This file is part of Polkadot.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Polkadot.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Cross-Consensus Message format asset data structures.
+//!
+//! This encompasses four types for representing assets:
+//! - `MultiAsset`: A description of a single asset, either an instance of a non-fungible or some amount of a fungible.
+//! - `MultiAssets`: A collection of `MultiAsset`s. These are stored in a `Vec` and sorted with fungibles first.
+//! - `Wild`: A single asset wildcard, this can either be "all" assets, or all assets of a specific kind.
+//! - `MultiAssetFilter`: A combination of `Wild` and `MultiAssets` designed for efficiently filtering an XCM holding
+//!   account.
+
+use super::{
+	Junction,
+	MultiLocation::{self, X1},
+};
+use alloc::{vec, vec::Vec};
+use core::cmp::Ordering;
+use parity_scale_codec::{self as codec, Decode, Encode};
+
+/// A general identifier for an instance of a non-fungible asset class.
+#[derive(Clone, Eq, PartialEq, Ord, PartialOrd, Encode, Decode, Debug)]
+pub enum AssetInstance {
+	/// Undefined - used if the NFA class has only one instance.
```

Comment:
```suggestion
	/// Undefined - used if the non-fungible asset class has only one instance.
```
unless the NFA acronym is defined somewhere easy to find